bot.py

```diff
@@ -1,545 +1,753 @@
+        if t == 'acodes': self.acodes.remove(code)
+        else: self.codes.remove(code)
+cfg = Config(CONFIG_FILE)
+
+# users #
+class Users:
+    def __init__(self, path):
+        try:
+            self.path = path
+            with open(self.path, 'r', encoding = 'utf-8') as c:
+                self.data = json.load(c)
+        except FileNotFoundError:
+            with open(self.path, 'w', encoding = 'utf-8') as c:
+                self.data = {}
+                self.update()
+    def update(self):
+        with open(self.path, 'w', encoding = 'utf-8') as c:
+            json.dump(self.data, c, indent = 4)
+    def getstatus(self, mess):
+        try:
+            if self.data[str(mess.from_user.id)]['key'] == ' ':
+                return Status.UNAUTH
+            elif self.data[str(mess.from_user.id)]['key'] == 'login':
+                return Status.LOGIN
+            elif self.data[str(mess.from_user.id)]['key'] == 'pass':
+                return Status.PASS
+            if self.data[str(mess.from_user.id)]['code'] != ' ':
+                return Status.MAYDAY
             else:
-                answer += f'{key_labels[key]}: {value}\n'
-    if 'lat' in data:
-        answer += f"[Точка на карте]({SERVICE_URL}/map?lat={data['lat']}&lon={data['lon']})\n"
-    else:
-        answer += '- - - - -\n'
-    return answer
-
-
-def formatpin(data):
-    key_labels = {
-        'name': 'Name',
-        'value': 'Pin',
-        'score': 'Score'
-    }
-    order = ['name', 'value', 'score']    # Order of values in the answer
-    copyable = ['value']   # Values that can be copied (monospaced font)
-    answer = ''
-    for element in order:
-        if (element in data) and data[element]:   # Check if value is not empty
-            key, value = element, data[element]
-            if key == 'score':
-                value = scoreformat(value)
-            if key in copyable:
-                answer += f'{key_labels[key]}: `{value}`\n'
+                return Status.AUTH
+        except:
+            code = mess.from_user.language_code
+            self.mod(mess = mess, login = ' ',
+                password = ' ', key = ' ',
+                code = ' ', acode = ' ',
+                language = 'ua' if code == 'ua' else 'ru' if code == 'ru' else 'us')
+            return Status.UNAUTH
+    def mod(self, mess = None, user_id = None, login = None, password = None,
+        key = None, code = None, acode = None, language = None):
+        user_id = str(mess.from_user.id) if user_id == None else user_id
+        self.data[user_id] = {
+        'login': login if login else self.data[user_id]['login'],
+        'pass': password if password else self.data[user_id]['pass'],
+        'key': key if key else self.data[user_id]['key'],
+        'code': code if code else self.data[user_id]['code'],
+        'acode': acode if acode else self.data[user_id]['acode'],
+        'language': language if language else self.data[user_id]['language']
+        }
```

Comment:
```suggestion
    def mod(self, mess = None, user_id = None, key = None, 
    code = None, acode = None, language = None):
        user_id = str(mess.from_user.id) if user_id == None else user_id
        self.data[user_id] = {
        'key': key if key else self.data[user_id]['key'],
        'code': code if code else self.data[user_id]['code'],
        'acode': acode if acode else self.data[user_id]['acode'],
        'language': language if language else self.data[user_id]['language']
        }
```
Оптимизация модельки